chapters/scoping.tex

```diff
@@ -135,6 +135,19 @@ \subsection{Composite Name Lookup}\doublelabel{composite-name-lookup}
 \autoref{redeclaration} for further explanations regarding looking inside
 partial packages.
 \end{nonnormative}
+\begin{example}
+Components and classes are part of the same name-space and thus a component may
+not have the same name as its class or the first part of the class-name as that
+would prevent looking up the name of the class.
+\begin{lstlisting}[language=modelica]
+model A
+  M M;    // Illegal, component 'M' prevents finding class 'M'
+  P.Q P;  // Illegal, component 'P' prevents finding package 'P'
+  .R R;   // Legal, see next section
+  S.Q Q;  // Legal
+end A;
+\end{lstlisting}
+\end{example}
```

Comment:
I like the example. It really clarifies. I wonder if the text and the example can be extended to also include cases of model C and model D in #2547.

Components and classes are part of the same name-space and thus a component may not have the same name as its class or the class of any other component in the same scope or the first part of the class-name....